src/Essential.Diagnostics.RollingFileTraceListener/Diagnostics/RollingFileTraceListener.cs

```diff
@@ -194,7 +225,7 @@ public string FilePathTemplate
         /// </summary>
         public override void Flush()
         {
-            rollingTextWriter.Flush();
+            rollingTextWriter.Flush(NewStreamOnError);
```

Comment:
Rather than pass on every write/flush, it would be better to have lazy creation of the rolling text writer (with double checking locking for multi-threaded), and pass NewStreamOnError into the constructor. e.g.

```
private RollingTextWriter RollingTextWriter { 
  get {
    if (rollingTextWriter == null)
    {
       lock (rollingTextWriterLock)
       {
          if (rollingTextWriter == null)
          {
             rollingTextWriter = new RollingTextWriter (... path ..., NewStreamOnError);
          }
        }
      }
    }
    return rollingTextWriter;
  }
}
```

Reply:
Fair enough, I commited the changes.